### PR TITLE
Inversion support for emp, nemp for sql; testcase emp, nemp

### PR DESCRIPTION
**Pull request by hayas**

### CODE REVIEW

File: src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java [hunk 1]
```diff
@@ -116,4 +119,43 @@ public void testPagination01() throws Exception
...
    }
 
+   @Test
+   public void testEmp01() throws Exception
+   {
+      Service service = service();
+      Response res = null;
+
+      res = service.get(url("orders?limit=5&emp(shipregion)"));
+      List<Map<String, Object>> resultsList = res.data().asList();
```
Comment (kfrankic) [inline src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java]: Replace with ```List<ObjectNode> list = res.data().asList()``` throughout.

**Review by kfrankic — CHANGES_REQUESTED** [review]
Make changes

File: src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java [hunk 2]
```diff
@@ -116,4 +119,43 @@ public void testPagination01() throws Exception
...
+   @Test
+   public void testEmp01() throws Exception
+   {
+      Service service = service();
+      Response res = null;
+
+      res = service.get(url("orders?limit=5&emp(shipregion)"));
+      List<ObjectNode> list = res.data().asList();
+      boolean assertion = true;
+      for (Map<String, Object> result : list)
```
Comment (kfrankic) [inline src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java]: You updated 'list' but failed to update Map<String, Object> to ObjectNode

File: src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java [hunk 3]
```diff
@@ -116,4 +118,43 @@ public void testPagination01() throws Exception
...
+   {
+      Service service = service();
+      Response res = null;
+
+      res = service.get(url("orders?limit=5&emp(shipregion)"));
+      List<ObjectNode> list = res.data().asList();
+      boolean assertion = true;
+      for (ObjectNode result : list)
+      {
+         if (result.get("shipregion") != null && result.get("shipregion") != "")
```
Comment (wellsb1) [inline src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java]: result.get("shipregion") != "" is NEVER valid Java.  You have to use .equals().  != checks object equality, not value equality.  

I would write this this way:

for(ObjectNode result : list)
{
  String shipregion = result.getString("shipregion");
  if(!Utils.empty(shipregion))
    fail("shipregion was supposed to be empty but was: '" + shipregion + "'");
}

File: src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java [hunk 4]
```diff
@@ -116,4 +118,43 @@ public void testPagination01() throws Exception
...
+         {
+            assertion = false;
+            break;
+         }
+      }
+      assertTrue(assertion);
+   }
+
+   @Test
+   public void testNemp01() throws Exception
```
Comment (wellsb1) [inline src/test/java/io/rocketpartners/cloud/action/rest/TestRestGetActions.java:156]: See comments for testEmp01() above.  Same thing here.

File: src/test/java/io/rocketpartners/cloud/action/sql/TestSqlQuery.java [hunk 5]
```diff
@@ -395,6 +395,16 @@ public void test1() throws Exception
                             "select *, `firstName` AS 'name' from table1 LIMIT 100", // 
                             "select *, `firstName` AS 'name' from table1 LIMIT 100"));//the dynamicsql and preparedsql should match without fields
 
+      tests.add(new RqlTest("emp(startYear)", //
```
Comment (wellsb1) [inline src/test/java/io/rocketpartners/cloud/action/sql/TestSqlQuery.java:398]: Nice job knowing to add these tests cases.  I did not explicitly ask for them.  This was the right thing to do.

**Review by wellsb1 — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by hayas** [comment]
Resolved all the requested changes

**Review by wellsb1 — APPROVED** [review]

**Pull request merged**